### PR TITLE
SEO: styles for suggested/max char count for front page meta description in wp-admin

### PR DESCRIPTION
**Pull request by samiff**
#### Changes proposed in this Pull Request:

- When users type over the suggested char length, a warning background
color is shown.
- When users type up to the max char length, a different warning
background is shown in addition to outputting that max chars has been
reached.

#### Jetpack product discussion

* Internal: p8oabR-DS-p2#comment-5149

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

* After pulling down these changes, in WP Admin navigate to `/wp-admin/admin.php?page=jetpack#/traffic`
* If SEO Tools are enabled, in the `Expand to edit your front page meta description.` section, you can type in a custom meta description. After typing 160 or more characters, a yellow background color should appear for the active character count. After typing 300 characters, the background should turn red, and text is show saying the max character count has been reached.

#### GIF

Apr-08-2021 10-58-40

### CODE REVIEW

File: projects/plugins/jetpack/_inc/client/traffic/style.scss [hunk 1]
```diff
@@ -116,6 +116,19 @@
 	&-count {
 		padding: 0 0 0.25rem 0.25rem;
```
Comment (sdixon194) [inline projects/plugins/jetpack/_inc/client/traffic/style.scss:117]:
```suggestion
		padding: 0.25rem;
```
Looks like the count isn't centered. The suggested fix here seems to fix it, but I'm not sure if that's the best method.

File: projects/plugins/jetpack/_inc/client/traffic/style.scss [hunk 2]
```diff
@@ -116,6 +116,19 @@
...
 	}
+
+	&-count-warn {
+		background-color: $alert-yellow;
```
Comment (sdixon194) [inline projects/plugins/jetpack/_inc/client/traffic/style.scss:121]: Should we match the styling for Calypso?

**Review by sdixon194 — CHANGES_REQUESTED** [review]

File: projects/plugins/jetpack/_inc/client/traffic/style.scss (hunk 2 above)
Comment (samiff) [reply projects/plugins/jetpack/_inc/client/traffic/style.scss:121]: Hmm I pulled these color vars from `projects/plugins/jetpack/_inc/client/scss/calypso-colors.scss`. There is also `projects/plugins/jetpack/scss/atoms/colors/_colors.scss`, but there aren't too many crayolas there. @jeherve might know a bit more about which color palettes to draw from here.

Comment (jeherve) [reply projects/plugins/jetpack/_inc/client/traffic/style.scss:121]: Let's go with `@import '../scss/variables/colors';` here if we can. The colors there are wp-admin colors, they'll probably fit better.

**Review by jeherve — CHANGES_REQUESTED** [review]

File: projects/plugins/jetpack/_inc/client/traffic/style.scss (hunk 2 above)
Comment (samiff) [reply projects/plugins/jetpack/_inc/client/traffic/style.scss:121]: Got it, looks like the color vars are actually the same here and we are already pulling in the `/scss/variables/colors` via the main style.scss file.

File: projects/plugins/jetpack/_inc/client/traffic/seo.jsx [hunk 3]
```diff
@@ -110,6 +110,16 @@ export const SEO = withModuleSettingsFormHelpers(
...
 
+			let frontPageMetaCharCountStyle;
+			if ( frontPageMetaDescription.length >= this.constants.frontPageMetaMaxLength ) {
+				frontPageMetaCharCountStyle = 'jp-seo-front-page-description-count-max';
+			} else {
+				frontPageMetaCharCountStyle =
+					frontPageMetaDescription.length > this.constants.frontPageMetaSuggestedLength
+						? 'jp-seo-front-page-description-count-warn'
+						: '';
+			}
```
[the remark this answers is not in the document]
Comment (samiff) [reply projects/plugins/jetpack/_inc/client/traffic/seo.jsx]: Thanks for the great tip Jeremy, using `classnames` does make things a lot cleaner to read here https://github.com/Automattic/jetpack/pull/19400/commits/13cdf788b14b4ec3ce76c1f9d257265004ed440b

**Review by sdixon194 — APPROVED** [review]
Nice, everything works well for me.

**Comment by jeherve** [comment]
Cherry-picked to `jetpack/branch-9.6` in 45477a57ad9e60febc9927c29a5336f3cfaf8ab1

**Pull request merged**